lib/bundling.js

```diff
@@ -60,12 +60,18 @@ function noop() {
  */
 function computeBundleId(obj, discriminatorFields) {
   var ids = [];
+  var hasIds = false;
   for (var i = 0; i < discriminatorFields.length; ++i) {
     var id = _.at(obj, discriminatorFields[i])[0];
     if (id === undefined) {
-      return undefined;
+      ids.push('');
```

Comment:
Oops, however, the current pattern does not capture the case that the data is structured, sigh. Uploaded a new patch to address this issue. This will result in an even more weird looking pattern, but I think this is still okay because of the reason above.